### ISSUE TITLE
REGEX inside lambda

### ISSUE THREAD

**Issue by undecided** [espruino/EspruinoWebIDE#226]
Hi! Got my banglejs at the beginning of the week, been loving it!
Have noticed a small issue when trying to write programs with regular expressions:
```
const storage = require('Storage');

const fileList = (regex) => storage.list(regex);
const appList = () => fileList(/\.info$/);

console.log(fileList(/\.info$/));

console.log(appList());
```
So I would expect both these `console.log`s to return the same array - looks like the following code is being generated:

```
function fileList(regex) {return storage.list(regex)}
function appList(undefined) {return fileList(REGEX)}
```
(as produced by `dump()`) but I can't see REGEX being defined anywhere. Don't have minification enabled, so I guess it just the transpiler at fault. 

Thought it would be easy to workaround with old-style anonymous functions - but they do exactly the same! Any thoughts?

**Comment by gfwilliams** [comment]
What have you got set in terms of minification in settings? If you disable it (the default) does it work?

**Comment by undecided** [comment]
Nope, minification was turned off for my code on both IDEs (enabled for modules, but turning that off had no effect). Turning it on has no effect either. 

It's weird that the code above works fine in a the emulator, but on my watch it fails. Found an even shorter example - this time, just using the console over bluetooth:

```
>/AAA/
=RegExp: {
  source: "AAA",
  lastIndex: 0 }
>function foobar() { var a = /AAA/; console.log(a) }
=function () { ... }
>foobar()
RegExp: {
  "source": "",
  "lastIndex": 0 }
```

so apologies, looks like that REGEX was just the way that the `dump()` command represents regex literals in functions. The Real issue seems to be that regex literals are not being handled correctly - which leads me to a workaround at least:

```
function foobar() { var a = new RegExp("AAA"); console.log(a) }
```

**Comment by gfwilliams** [comment]
Thanks - actually I can just move this issue itself which is probably better.

Please can you try `E.setFlags({pretokenise:0})` at the start of the code you upload? I bet it's a bug related to pretokenisation.

For instance this will fail in the emulator:

```
E.setFlags({pretokenise:1})
function foobar() { var a = /AAA/; console.log(a) }
foobar()
```

It might be worth turning off pretokenise for Bangle.js now - now we're executing from flash there's not much benefit.
